### ISSUE TITLE
Template fails with BadRequest error for Azure roleAssignments

### ISSUE THREAD

**Issue by difanzhao** [F5Networks/f5-azure-arm-templates#231]
I think this is similar to issue ESECLDTPLT-2041. However, I should be using a version that has fixed the bug but I am still getting a similar error.

Template: f5-azure-arm-templates/supported/failover/same-net/via-api/n-nic/existing-stack/byol/azuredeploy.json
Version: 9.6.0.0

I am getting this Azure error when trying to use this template to deploy
```
{
    "status": "Failed",
    "error": {
        "code": "RoleAssignmentUpdateNotPermitted",
        "message": "Tenant ID, application ID, principal ID, and scope are not allowed to be updated."
    }
}
```

I got the same error twice, one for each VM in the HA pair. The VMs are still built successfully. I can manually assign the contributor role for the Resource Group to the VMs (or their system-assigned management identity) but I can't do it with the template.

Every time I deploy, in the Azure Deployment details I see exactly the same GUID value for the Microsoft.Authorization/roleAssignments resource. Is it not supposed to be random so different every time?

I think this is due to multiple deployments with the same template and parameter files in the same RG. Could you investigate? Thanks!

**Comment by difanzhao** [comment]
Hi Shyawn, sorry for the late response... yes - your workaround works. If I deleted those orphaned role assignments, I don't get the same error. The parameter file is attached. Had to rename it to "txt" file in order to attach it here
[f5_azuredeploy.parameters.txt](https://github.com/F5Networks/f5-azure-arm-templates/files/7489517/f5_azuredeploy.parameters.txt)
.

**Comment by shyawnkarim** [comment]
I've filed a bug for this issue, internal ID ESECLDTPLT-2892, to investigate why this error is still occurring. Thanks for reporting and for sharing your parameters file.

**Comment by mikeshimkus** [comment]
@difanzhao Hi, I'm having a look at this.

Can you tell me what you change in the parameters file when you create the 2nd deployment in the same resource group? If you can share the parameters for each deployment, that would help.

**Comment by mikeshimkus** [comment]
@difanzhao Following up, I was able to reproduce the error by submitting two deployments where:

- dnsLabel and instanceName parameter values were the same in both deployments
- roleNameGuid was the same in both deployments (not the literal "[newGuid()]", but the ARM output of newGuid() from deployment 1 was resubmitted as value of roleNameGuid for deployment 2)

If you are intending to create a new deployment, you should _always_ have a unique value for dnsLabel; this determines resource name uniqueness for resources that require it. Since the role assignment name is a based on a concatenation of dnsLabel and instanceName, this should prevent the error in the future.

I was also able to reproduce the error by _redeploying_ a template using the same values, but only after deleting the BIG-IP virtual machines.

If you are updating an existing deployment, both dnsLabel and roleNameGuid can stay the same _as long as the virtual machines have not been deleted_. Azure will recreate the virtual machine with a new principal ID, then try to modify the existing role assignment to use it which will also result in this error.

**Comment by shyawnkarim** [comment]
Closing due to age. These legacy templates are now in maintenance mode and are being replaced by our next-generation templates available in the [Cloud Templates 2.0 GitHub repo](https://github.com/F5Networks/f5-azure-arm-templates-v2).
